Libraries/Image/RCTImageLoader.m

```diff
@@ -480,6 +470,18 @@ - (RCTImageLoaderCancellationBlock)loadImageWithURLRequest:(NSURLRequest *)image
   __block void(^cancelLoad)(void) = nil;
   __weak RCTImageLoader *weakSelf = self;
 
+  // Load handlers are executed before the generic caching logic so they are responsible
+  // for caching the images if necessary.
+  id<RCTImageURLLoader> loadHandler = [self imageURLLoaderForURL:imageURLRequest.URL];
```

Comment:
Maybe passing an additional parameter to the completion block of RCTImageURLLoader so loaders can decide if the result will be cached or not.